doc/calling-javascript.md

```diff
@@ -39,97 +41,56 @@ The core type hierarchy is as follows:
      |           +- js.ThisFunction1[-T0, -T1, +R]
      |           +- ...
      |           +- js.ThisFunction21[-T0, ..., -T21, +R]
-     +- js.prim.Number
-     +- js.prim.Boolean
-     +- js.prim.String
-     +- js.prim.Undefined
-
-
-Note that the types in the `prim` package should not directly be used,
-since they have a direct correspondance in Scala (see the
-[interoperability](./js-interoperability.html) page). The reason they
-exist is that for example a `scala.Double` can be stored in a variable
-of type `js.Any` (through implicit conversion to `js.prim.Number`).
-
-A value of any of these types is encoded as is in JavaScript, without wrapping.
-Even when such a value is assigned to a `val` of type `scala.Any` or of a
-generic type, there is no boxing. E.g., a `js.Array[js.Date]` is a JavaScript
-`Array` which contains JavaScript `Date`'s at runtime.
-
-These types have all the fields and methods available in the JavaScript API.
-
-There are implicit conversions from corresponding Scala types and back:
-
-<table class="table table-bordered">
-  <thead>
-    <tr><th>Scala type</th><th>JavaScript type</th></tr>
-  </thead>
-  <tbody>
-    <tr><td>Byte<br/>Short<br/>Int<br/>Long<br/>Float<br/>Double</td><td>js.prim.Number</td></tr>
-    <tr><td colspan="2">(from js.prim.Number to Double only)</td></tr>
-    <tr><td>Boolean</td><td>js.prim.Boolean</td></tr>
-    <tr><td>java.lang.String</td><td>js.prim.String</td></tr>
-    <tr><td>Unit</td><td>js.prim.Undefined</td></tr>
-    <tr><td>Array[A]</td><td>js.Array[A]</td></tr>
-    <tr><td rowspan="2">FunctionN[T1, ..., TN, R]</td><td>js.FunctionN[T1, ..., TN, R]</td></tr>
-    <tr>                                              <td>js.ThisFunction{N-1}[T1, ..., TN, R]</td></tr>
-  </tbody>
-</table>
-
-### Remarks
-
-There is no type `js.Null`, because `scala.Null` can be used in its stead with
-the appropriate semantics.
+     +- js.Dictionary[A]
 
-`isInstanceOf[T]` is supported for _classes_ inheriting from `js.Object`, e.g.,
-`js.Date`, `js.Array[_]`, `js.Object` itself, and is implemented with an
-`instanceof` test.
+Note that most of these types are similar to standard Scala types. For example,
+`js.Array[A]` is similar to `scala.Array[A]`, and `js.FunctionN` is similar to
+`scala.FunctionN`.
+However, they are not completely equivalent, and must not be confused.
 
-`isInstanceOf[T]` for `T` being `js.prim.Number`, `js.prim.Boolean`,
-`js.prim.String`, or `js.prim.Undefined`, is supported and is
-implemented with a `typeof` test. However, it should be avoided in
-preference of `T` being `Double`, `Boolean`, `String` or `Unit`.
+With the exception of `js.Array[A]` and `js.Dictionary[A]`, these types have
+all the fields and methods available in the JavaScript API.
+The collection types feature instead the standard Scala collection API, so that
```

Comment:
The collection types feature <del>instead</del> the standard Scala collection API <ins>instead</ins>.